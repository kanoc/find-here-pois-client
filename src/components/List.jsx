import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import ListItem from './ListItem'
import { selectItem } from "../store/actions";

const ListContainer = styled.div`
    display: flex;
    flex-direction: column;
    text-align: left;
    width: 408px;
    box-shadow: 0 4px 10px -2px rgba(0, 0, 0, 0.5);
    z-index: 5;
    overflow-y: auto;
`;

const ListHeader = styled.span`
    font-weight: bold;
    padding: 10px 18px 20px 24px;
`;

const Divider = styled.hr`
    width: 100%;
    height: 0;
    margin:0;
    padding:0;
    border: 0;
    border-bottom: 1px solid #e9e9e9;
    overflow: hidden;
`;

const mapStateToProps = state => ({
    hotels: state.hotels,
    selectedId: state.selectedId
});

const mapDispatchToProps = dispatch => ({
    selectHotel: itemId => dispatch(selectItem(itemId)),
});

const ConnectedList = ({ hotels, selectedId, selectHotel }) => {

    const onItemClick = (itemId) => {
        selectHotel(itemId)
    };

    const headerText = hotels.length >= 1
        ? "An exquisite selection of wonderful hotels in the area:"
        : "No hotels found in the area. Zoom in/out or move the map around!";

    return (
        <ListContainer>
            <ListHeader>{headerText}</ListHeader>
            {hotels.map(item => (
                <React.Fragment key={item.id}>
                    <Divider/>
                    <ListItem key={item.id} item={item} isSelected={item.id === selectedId} onItemClick={onItemClick}/>
                </React.Fragment>
            ))}
        </ListContainer>
    );
};


export default connect(mapStateToProps, mapDispatchToProps)(ConnectedList);
