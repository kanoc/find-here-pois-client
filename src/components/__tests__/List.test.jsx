import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import List from '../List';

const mockStore = configureStore([]);

describe('ConnectedList component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            bbox: [],
            selectedId: '',
            hotels: [
                {
                    id: '123',
                    title: 'Foo1',
                    rating: 1,
                },
                {
                    id: '456',
                    title: 'Foo2',
                    rating: 2,
                },
            ],
        });
        component = renderer.create(
            <Provider store={store}>
                <List/>
            </Provider>
        );
    });

    it('renders correctly', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
