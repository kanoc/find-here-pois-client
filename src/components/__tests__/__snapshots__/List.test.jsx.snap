// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedList component renders correctly 1`] = `
<div
  className="sc-bwzfXH eTBoOe"
>
  <span
    className="sc-htpNat gJDJnL"
  >
    An exquisite selection of wonderful hotels in the area:
  </span>
  <hr
    className="sc-bxivhb liDJGs"
  />
  <div
    className="sc-bdVaJa hpfjMZ"
    onClick={[Function]}
  >
    <div>
      Foo1
    </div>
    <div>
      <i>
        Rating: 
        1
      </i>
    </div>
  </div>
  <hr
    className="sc-bxivhb liDJGs"
  />
  <div
    className="sc-bdVaJa hpfjMZ"
    onClick={[Function]}
  >
    <div>
      Foo2
    </div>
    <div>
      <i>
        Rating: 
        2
      </i>
    </div>
  </div>
</div>
`;
